### DISCUSSION TITLE
Shell navigation parameters VS. Page lifecycle events

### DISCUSSION THREAD

**Discussion by the-gozo** [dotnet/maui#26646] [Q&A]
How to ensure that parameters passed during Shell navigation are set before Page's NavigatedTo or Appearing event are fired? I want to use those parameters during or after page initialization in some setup logic, without user interaction. Is it possible? If not, what is the best practice for similar use cases?

**Comment by bhavanesh2001** [comment]
You can define a filed backed property for storing the parameters passed during the navigation. 

You will know exactly once the value of the property is set, then you can initialize you setup logic.

Example :
```csharp
private string _parameter;
public string Parameter
{
  get
  {
    return _parameter;
  }
  set
  {
     _parameter= value;
     if (!string.IsNullOrEmpty(value))
     {
       // Initialize your logic
     }
  }
}
```
